### PR TITLE
Remove all instances of `== Sorts.Irrelevant`

### LINKED ISSUE #18845: Terms in SProp not found convertible (sort polymorphism)
### Description of the problem

Two terms in SProp are not found convertible (wrong relevance computation).

### Small Coq file to reproduce the bug

```coq
Set Universe Polymorphism.

Inductive Box@{s|u|} (A:Type@{s|u}) := box (_:A).

Definition t1@{s|u|} (A:Type@{s|u}) (x y : A) := box _ x.
Definition t2@{s|u|} (A:Type@{s|u}) (x y : A) := box _ y.

Check fun A:SProp => eq_refl : t1 A = t2 A. (* What is in "test-suite/success/sort_polymorphism.v" *)
Check fun (A : SProp) => (fun (x : t1 A = t2 A) => x) eq_refl. (* Slight variation *)
```

### Version of Coq where this bug occurs

8.19.1, master (`d2c3b06f66`)

### Last version of Coq where the bug did not occur

_No response_

### PR DESCRIPTION
**Pull request by yannl35133**
Fixes / closes #18845

I'm quite unsure of what I did, please review carefully.

- [x] Added / updated **test-suite**.

### CODE REVIEW

File: engine/uState.ml [hunk 1]
```diff
@@ -406,6 +406,12 @@ let nf_relevance uctx r = match r with
     else if QVar.equal q q' then r
     else Sorts.RelevanceVar q'
 
+let is_irrelevant uctx r =
```
[the remark this answers is not in the document]
Comment (yannl35133) [reply engine/uState.ml:409]: It's used in `Evd` and `Evarutil`, so should I move it to the former?

[the remark this answers is not in the document]
Comment (ppedrot) [reply engine/uState.ml:409]: I'd say just Evd then, but maybe with a more specific name like `is_irrelevant_relevance` or something.

File: engine/evd.ml [hunk 2]
```diff
@@ -950,18 +950,10 @@ let mkLEvar sigma (evk, args) =
...
   let evar_irrelevant (evk, _) = match find sigma evk with
-  | EvarInfo evi ->
-    begin match evi.evar_relevance with
-    | Sorts.Relevant -> false
-    | Sorts.Irrelevant -> true
-    | Sorts.RelevanceVar q -> qvar_irrelevant q
-    end
-  | exception Not_found -> true
+  | EvarInfo evi -> UState.is_irrelevant sigma.universes evi.evar_relevance
+  | exception Not_found -> true (* XXX: Why this default ?? *)
```
Comment (SkySkimmer) [inline engine/evd.ml:956]: It's wrong indeed. The function used to be the flipped `evar_relevant` and this wasn't updated in b2d2e6f8787a593fc413a61252774692aaa03052.
(also probably should be anomaly)

Comment (ppedrot) [reply engine/evd.ml:956]: AFAIR you cannot put an anomaly there because many people play fast and loose with evarmaps (hm, hm).

Comment (yannl35133) [reply engine/evd.ml:956]: I guess that answers my question regarding a possble check in (to be moved) `UState.is_irrelevant`

File: engine/evd.ml [hunk 3]
```diff
@@ -948,20 +948,18 @@ let mkLEvar sigma (evk, args) =
   let args = List.fold_right2 fold (evar_filtered_context evi) args SList.empty in
   mkEvar (evk, args)
 
+let is_relevance_irrelevant sigma r =
+  match UState.nf_relevance sigma.universes r with
+  | Irrelevant -> true
+  | Relevant -> false
+  | RelevanceVar _ -> false (* Should be an anomaly *)
```
Comment (ppedrot) [inline engine/evd.ml:955]: The comment is wrong: there can be undefined sort variables.

Comment (yannl35133) [reply engine/evd.ml:955]: I changed it, but I'm gonna wait for more significant changes before pushing

Comment (ppedrot) [reply engine/evd.ml:955]: Ping. The rest of the PR is fine to me.

File: engine/evd.ml [hunk 4]
```diff
@@ -948,20 +948,18 @@ let mkLEvar sigma (evk, args) =
   let args = List.fold_right2 fold (evar_filtered_context evi) args SList.empty in
   mkEvar (evk, args)
 
+let is_relevance_irrelevant sigma r =
+  match UState.nf_relevance sigma.universes r with
+  | Irrelevant -> true
+  | Relevant -> false
+  | RelevanceVar _ -> false (* Should check whether the variable is registered *)
```
Comment (SkySkimmer) [inline engine/evd.ml:955]: no need for a comment here
we already have the FIXME comment in ustate qstate.repr

**Review by ppedrot — APPROVED** [review]

**Comment by ppedrot** [comment]
@coqbot run full ci

**Comment by ppedrot** [comment]
@SkySkimmer are you OK with the PR in its current state? If yes I can merge right away.

**Review by SkySkimmer — APPROVED** [review]

**Comment by ppedrot** [comment]
@coqbot merge now

**Pull request merged**